packages/rum-core/src/common/config-service.js

```diff
@@ -198,17 +197,23 @@ class Config {
    */
   validate(properties = {}) {
     const requiredKeys = ['serviceName', 'serverUrl']
+    const allKeys = Object.keys(this.config)
     const errors = {
       missing: [],
-      invalid: []
+      invalid: [],
+      unknown: []
     }
     /**
-     * Check when required keys are missing
+     * Check when required keys are missing or unknown keys found
      */
     Object.keys(properties).forEach(key => {
       if (requiredKeys.indexOf(key) !== -1 && !properties[key]) {
         errors.missing.push(key)
       }
+
+      if (allKeys.indexOf(key) < 0) {
```

Comment:
nit:

What do you think about checking against `!== -1` rather than `< 0`? 

(Proposing this to use the same style as others indexOf in the codebase)

Reply:
yes, I'm all about consistency so I like your suggestion!! will update 👍